src/OpenTelemetry/Metrics/IDataPoint.cs

```diff
@@ -0,0 +1,37 @@
+// <copyright file="IDataPoint.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Metrics
+{
+    public interface IDataPoint
+    {
+        public DateTimeOffset Timestamp { get; }
```

Comment:
I don't think you want the `public` on these. I think that makes it default implementation instead of interface definition 😄 Could be fine, not totally sure TBH.

Reply:
Humm, I don't know either. Will need to do a little research on this.

Reply:
I read up on this, and we are allowed to have accessibility modifiers, and in this case, we are not defining any default interface implementation. I also checked on benchmark with and without accessibility modifiers, and it made no difference.

But, I'll remove the modifiers to prevent confusions going forward.

```
Interface members are public by default, and you can explicitly specify accessibility modifiers, such as public, protected, internal, private, protected internal, or private protected. A private member must have a default implementation.
```
```
Beginning with C# 8.0 on .NET Core 3.0, you can define an implementation when you declare a member of an interface. The most common scenario is to safely add members to an interface already released and used by innumerable clients.
```